src/cursor/cur_hs.c

```diff
@@ -582,8 +587,12 @@ __curhs_search_near(WT_CURSOR *cursor, int *exactp)
         WT_ERR(__curhs_prev_visible(session, hs_cursor));
     }
 
-done:
+    if (0) {
 err:
+        WT_TRET(cursor->reset(cursor));
+    }
+
+done:
```

Comment:
WT_NOTFOUND case, is jumping to done: but when we return WT_NOTFOUND, cursor should not be positioned, so that handling should be similar to err handling.